website/frontend/src/components/pages/Programs/children/All/All.tsx

```diff
@@ -79,35 +52,13 @@ export const All: VFC = () => {
       </div>
       <ProgramsLegend />
       <div className={styles.allProgramsList}>
-        {(programs &&
-          programsCount &&
-          programs.map((item: ProgramModel) => {
-            if (item.name && item.name !== 'name.wasm') {
-              return <UserProgram program={item} handleOpenForm={handleOpenForm} key={item.id} />;
-            }
-            return (
-              <div className={styles.allProgramsItem} key={item.id}>
-                <p className={styles.allProgramsItemHash}>{item.id}</p>
-                <div className={styles.programsListBtns}>
-                  <Link to={`/send/message/${item.id}`} className={styles.allProgramsItemSendMessage}>
-                    <img src={MessageIcon} alt="Send message to program" />
-                  </Link>
-                  <button
-                    className={styles.allProgramsItemUpload}
-                    type="button"
-                    onClick={() => handleOpenForm(item.id, item.name)}
-                  >
-                    <img src={UploadIcon} alt="upload-program" />
-                  </button>
-                </div>
-              </div>
-            );
-          })) ||
-          null}
+        {programs.map((program: ProgramModel) => (
+          <UserProgram key={program.id} program={program} disabledMeta={publicKeyRaw !== program.owner} />
```

Comment:
```suggestion
          <UserProgram key={program.id} program={program} />
```
I'd say no need for additional prop, since we're already have `program` prop and can do this check inside

Reply:
I don't think that's right. It's a simple component